pkg/network/tracer/connection/fentry/probes.go

```diff
@@ -0,0 +1,140 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package fentry
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+)
+
+const (
+	// inetCskListenStop traces the inet_csk_listen_stop system call (called for both ipv4 and ipv6)
+	inetCskListenStop = "fentry/inet_csk_listen_stop"
+
+	// tcpConnect traces the connect() system call
+	tcpConnect = "fentry/tcp_connect"
+	// tcpFinishConnect traces tcp_finish_connect() kernel function. This is
+	// used to know when a TCP connection switches to the ESTABLISHED state
+	tcpFinishConnect = "fentry/tcp_finish_connect"
+
+	// tcpSendMsgReturn traces the return value for the tcp_sendmsg() system call
+	tcpSendMsgReturn = "fexit/tcp_sendmsg"
+
+	// tcpSetState traces the tcp_set_state() kernel function
+	tcpSetState = "fentry/tcp_set_state"
+
+	// tcpRecvMsgReturn traces the return value for the tcp_recvmsg() system call
+	tcpRecvMsgReturn = "fexit/tcp_recvmsg"
+	// tcpClose traces the tcp_close() system call
+	tcpClose = "fentry/tcp_close"
+	// tcpCloseReturn traces the return of tcp_close() system call
+	tcpCloseReturn = "fexit/tcp_close"
+
+	// We use the following two probes for UDP
+	udpRecvMsgReturn   = "fexit/udp_recvmsg"
+	udpSendMsgReturn   = "fexit/udp_sendmsg"
+	udpSendSkb         = "kprobe/udp_send_skb"
+	udpv6RecvMsgReturn = "fexit/udpv6_recvmsg"
+	udpv6SendMsgReturn = "fexit/udpv6_sendmsg"
+	udpv6SendSkb       = "kprobe/udp_v6_send_skb"
+
+	// udpDestroySock traces the udp_destroy_sock() function
+	udpDestroySock = "fentry/udp_destroy_sock"
+	// udpDestroySockReturn traces the return of the udp_destroy_sock() system call
+	udpDestroySockReturn = "fexit/udp_destroy_sock"
+
+	// tcpRetransmit traces the return value for the tcp_retransmit_skb() system call
+	tcpRetransmitReturn = "fexit/tcp_retransmit_skb"
+
+	// inetCskAcceptReturn traces the return value for the inet_csk_accept syscall
+	inetCskAcceptReturn = "fexit/inet_csk_accept"
+
+	// inetBindRet is the kretprobe of the bind() syscall for IPv4
+	inetBindRet = "fexit/inet_bind"
+	// inet6BindRet is the kretprobe of the bind() syscall for IPv6
+	inet6BindRet = "fexit/inet6_bind"
+
+	// sockFDLookupRet is the kretprobe used for mapping socket FDs to kernel sock structs
+	sockFDLookupRet = "fexit/sockfd_lookup_light"
+
+	// doSendfileRet is the kretprobe used to trace traffic via SENDFILE(2) syscall
+	doSendfileRet = "fexit/do_sendfile"
+)
+
+var programs = map[string]string{
+	doSendfileRet:        "do_sendfile_exit", // no
```

Comment:
Can you add a comment explaining why this and `sockFDLookupRet` are not used?

Reply:
See https://datadoghq.atlassian.net/browse/NET-2828?focusedCommentId=821557

Reply:
Gotcha. could you please update your comment to:
> No alternative present. Will have to use one or more hooks to get equivalent functionality. Only affects do_sendfile **and HTTPS monitoring (OpenSSL/GnuTLS/GoTLS).**

?